### PR TITLE
fix: dedupe handleVoteUpdateByRepo function

### LINKED ISSUE #271: Bug: dedupe handleVoteUpdateByRepo function
### Describe the bug

Any part of the website that can be used to vote requires posthog and as such the `handleVoteUpdateByRepo` function should be deduplicated and moved to the lib folder.

### Steps to reproduce

Duplicated code exists in:
- [ ] `PostList.tsx`
- [ ] `PostGrid.tsx`
- [ ] `HotRepositories.tsx`

### Affected services

hot.opensauced.pizza

### Platforms

_No response_

### Browsers

_No response_

### Environment

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by SunGoldTech**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR replaces duplicate codes with a single reusable one.

## Related Tickets & Documents

fixes #271 

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -2,14 +2,12 @@ module.exports = {
   parser: "@typescript-eslint/parser",
```
Comment (NsdHSO) [inline .eslintrc.cjs:2]: I don't understand why you reformat this file, please revert.

**Review by NsdHSO — COMMENTED** [review]
Please added only what you did.

File: .eslintrc.cjs (hunk 1 above)
Comment (SunGoldTech) [reply .eslintrc.cjs:2]: Ok..

**Review by 0-vortex — APPROVED** [review]
LGTM! Regarding the eslint formatting, was this done by the IDE automatically? (like vsc eslint/prettier autoconf) 

I will add this file to the eslint ignore array in a future update just to be sure. 

Will test the unknown consequence and merge this afterwards (might take a day tho)

Thank you for another great contribution, feel free to grab another one if you feel like it! 🙇 🍕 🚀

**Comment by SunGoldTech** [comment]
> LGTM! Regarding the eslint formatting, was this done by the IDE automatically? (like vsc eslint/prettier autoconf)
> 
> I will add this file to the eslint ignore array in a future update just to be sure.
> 
> Will test the unknown consequence and merge this afterwards (might take a day tho)
> 
> Thank you for another great contribution, feel free to grab another one if you feel like it! 🙇 🍕 🚀

it was auto formatted by my IDE automatically... thanks for the review. i'm still new to typescript though :).

**Pull request merged**